core/lib/vm/src/tracers/traits.rs

```diff
@@ -11,16 +11,14 @@ use crate::types::outputs::VmExecutionResultAndLogs;
 use crate::{Halt, VmExecutionStopReason};
 
 /// Run tracer for collecting data during the vm execution cycles
-pub trait ExecutionProcessing<S: WriteStorage, H: HistoryMode>:
-    DynTracer<S, H> + ExecutionEndTracer<H>
-{
+pub trait VmTracer<S: WriteStorage, H: HistoryMode>: DynTracer<S, H> {
     fn initialize_tracer(&mut self, _state: &mut ZkSyncVmState<S, H>) {}
-    fn before_cycle(&mut self, _state: &mut ZkSyncVmState<S, H>) {}
-    fn after_cycle(
+    fn finish_cycle(
```

Comment:
can you add a comment - when is this method executed? after each op code?

Reply:
i've added comments when it's called